nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/themes/nifi.scss

```diff
@@ -49,355 +53,100 @@ $material-primary-light-palette: (
-        A700: rgba(black, 0.87),
-    )
-);
-
-// The $nifi-canvas-light-palette defines the ACCENT palette for all flow designer canvas components that make up the NiFi canvas theme used throughout Apache NiFi
-$nifi-canvas-accent-light-palette: (
-    // 50 -> 900 are the PRIMARY colors defined by https://m2.material.io/design/color/the-color-system.html#tools-for-picking-colors for primary color #52bf7e
-    50: #e6f6ec,
-    100: #c3e8d0, // "lighter" hue for this palette.
-    200: #9dd9b2, //.running
-    300: #73ca94, //.backpressure-percent
-    400: #52bf7e, // color="primary" Default hue for this palette. Also .up-to-date
-    500: #2cb367,
-    600: #1A9964, //.version-control
-    700: #016131, // "darker" hue for this palette Also .bulletin-normal
-    800: #0000ff, //.endpoint, g.process-group.drop rect.border
-    900: #00ff00, //.connectable-destination, .connector.connectable
-
-    // A100 - A700 are the ANALOGOUS colors but are more customized. These colors are used to highlight, warn, denote midpoints and labelpoints, etc
-    A100: #ffef85, //.selected
-    A200: #f8bf47, //.invalid, .is-missing-port, circle.context
-    A400: #bda500, //.backpressure-percent-warning, .bulletin-warn, .backpressure-percent.warning, text.run-status-icon
-    A700: #ffcc00, //g.connection.selected rect.border, .connection-selection-path, .midpoint, .labelpoint
-
-    // These are the $nifi-canvas-accent-light-palette PRIMARY palette contrast colors. These color do not really get defined by https://m2.material.io/design/color/the-color-system.html#tools-for-picking-colors.
-    // Instead if we look to the Angular Material provided palettes we see that these fields are typically rgba(black, 0.87) or white. These values are particularly important
-    // for light mode and dark mode as these values set the colors for the text when displayed against the primary background on a button, badge, chip, etc.
-    //
-    // NOTE: Care should be taken here to ensure the values meet accessibility standards.
-    //
-    // NOTE: When creating the Material palette definition mat.define-palette($nifi-canvas-accent-light-palette, 400, 100, 700);
-    // Since 400 is the default the contrast-400 will be used as the default text color in some cases.
-    contrast: (
-        50: rgba(black, 0.87),
-        100: rgba(black, 0.87),
-        200: rgba(black, 0.87),
-        300: rgba(black, 0.87),
-        400: rgba(black, 0.87),
-        500: rgba(black, 0.87),
-        600: #ffffff,
-        700: #ffffff,
-        800: #ffffff,
-        900: #ffffff,
-        A100: rgba(black, 0.87),
-        A200: rgba(black, 0.87),
-        A400: rgba(black, 0.87),
-        A700: rgba(black, 0.87),
+        50: $on-surface-dark,
+        100: $on-surface-dark,
+        200: $on-surface-dark,
+        300: $on-surface-dark, 
+        400: $on-surface-dark,
+        500: $on-surface-dark,
+        600: #004849, // We use a custom contrast color for the primary default color.
```

Comment:
I see that you are using this color to calculate the nifi-navigation toolbar background color as well as the draggable icons, text, etc.  but according to everything I know about Material Design and the Angular Material implementation of it (which isn't everything but I do have quite a bit of experience with it) this contrast color should not be dark like this. It should pretty much be white. When I run the suggestion I recently made (which I think follow more closely with the Angular Material example palettes) in the Angular Material demo app I see the toolbars look like this:

Light Mode:
<img width="1727" alt="Screenshot 2024-03-08 at 10 04 20 PM" src="https://github.com/apache/nifi/assets/6797571/5ab10892-5967-4357-8a1e-40cff0f73310">

Dark Mod
<img width="1728" alt="Screenshot 2024-03-08 at 10 04 11 PM" src="https://github.com/apache/nifi/assets/6797571/408028eb-b3d6-4dc9-9e19-9bd1ea0ba7bb">
e:

I really like and appreciate you trying to make use of #004849 as an accent color in the new nifi ui as it is such a prevalent and iconic color in the legacy nifi ui. Do you think there is an alternative approach we can use for the nifi-navigation toolbar that does not require such a seemingly wrong contrast color like this?

Reply:
Thanks for the update.